liquidctl/driver/nzxt_smart_device.py

```diff
@@ -121,15 +118,77 @@
 _MAX_DUTY = 100
 _READ_ENDPOINT = 0x81
 _READ_LENGTH = 21
+_READ_LENGTH_V2 = 60
 _WRITE_ENDPOINT = 0x1
 _WRITE_LENGTH = 65
```

Comment:
Is the `_WRITE_LENGTH` correct for the Smart Device V2?

I mean, on macOS it will probably work anyway since we're using HIDAPI.  But this might break successive calls on Windows/Linux, which currently default to use libusb.

---

I'm not sure it will be able to show the HID descriptors on macOS, but can you install the real¹ `lsusb` and run `lsusb -v -d 1e71:2006` as root?

¹ [(StackOverflow) macos - Is there an equivalent of lsusb for OS X (Comment)](https://stackoverflow.com/a/51579385/2801377), possibly outdated.

Reply:
CAM writes 64 bytes according to captured USB traffic. Will test on Mac and Windows.

Reply:
In that case `_WRITE_LENGTH` is probably correct, because it (IIRC) has to include the report ID.  I'll check this on my end as well.

Reply:
I'd like to move _READ_LENGTH and _WRITE_LENGTH to class-level and thus avoid the use of "_READ_LENGTH_V2".

The V1 driver will use:
- _READ_LENGTH = 21
- _WRITE_LENGTH = 65

The V2 driver will use:
- _READ_LENGTH = 60
- _WRITE_LENGTH = 64

Reply:
Scratch what I just wrote.

`_WRITE_LENGTH` also needs to be customized between driver versions.  I think the best solution for this is to make it (and also `_READ_LENGTH`) class variables.

This will also avoid potential mistakes from having both `_READ_LENGTH` and `_READ_LENGTH_V2`.

Reply:
Ops, GitHub didn't show me your message in time.  Yeah, please do that! : )